packages/consent/consent-tools/src/domain/create-wrapper.ts

```diff
@@ -90,10 +89,15 @@ export const createWrapper: CreateWrapper = (createWrapperOptions) => {
 
         if (!allCategories.length) {
           // No configured integrations found, so no categories will be sent (should not happen unless there's a configuration error)
-          return undefined
+          throw new ValidationError(
+            'Invariant: No consent categories defined in Segment',
+            []
+          )
```

Comment:
Do we handle this Exception or is this bubbled up all the way to the user?

Just curious. I'm ok with this preventing any operation as consent is not defined at this point, but something to call out in the documentation so that people set their categories first before enabling consent in their clients.

Reply:
> Do we handle this Exception or is this bubbled up all the way to the user?
> 
> Just curious. I'm ok with this preventing any operation as consent is not defined at this point, but something to call out in the documentation so that people set their categories first before enabling consent in their clients.


This error stops the pipeline -- it's async, so the error is exposed to the user via the usual mechanism (I think like any other plugin, it's logged in context.failedDelivery()). I don't think it bubbles up completely.

I think that's a reasonable thing to document (we are still in private beta, so don't have to worry about that). Categories need to be set regardless -- since we're validating categories on first load -- this feature does not change that behavior.